lib/api/2.0/ibms.js

```diff
@@ -0,0 +1,118 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var injector = require('../../../index.js').injector;
+var controller = injector.get('Http.Services.Swagger').controller;
+var schemaApiService = injector.get('Http.Api.Services.Schema');
+var nameSpace = '/api/2.0/ibms/definitions/';
+var waterline = injector.get('Services.Waterline');
+var eventsProtocol = injector.get('Protocol.Events');
+var Errors = injector.get('Errors');
+var _ = injector.get('_'); // jshint ignore:line
+
+
+// GET /api/2.0/ibms
+var ibmsGet = controller(function(req) {
+    return waterline.ibms.find(req.query);
+});
+
+// PUT /api/2.0/ibms
+var ibmsPut = controller({success: 201}, function(req) {
+    var nodeId = req.swagger.params.body.value.nodeId;
+    delete req.swagger.params.body.value.nodeId;
+    return waterline.ibms.upsertByNode(nodeId, req.swagger.params.body.value);
+});
+
+// GET /api/2.0/ibms/identifier
+var ibmsGetById = controller(function(req) {
+    return waterline.ibms.needByIdentifier(req.swagger.params.identifier.value);
+});
+
+// PATCH /api/2.0/ibms/identifier
+var ibmsPatchById = controller( function(req) {
+    var ibmId = req.swagger.params.identifier.value;
+    var values = req.swagger.params.body.value;
+    return waterline.ibms.needByIdentifier(ibmId)
```

Comment:
@anhou @srinia6 The duplicated code is in `ibmsGet`, `ibmsPut`, `ibmsGetById`, and `ibmsGetAll`.  These are all basically one-liners, does it really make sense to create another layer of abstraction for this?  The two verbs with some substance to them (patch and delete) are not common anyway.  They could be moved to a service, but that does not avoid duplication.